### PR TITLE
[plugin/admission_test] fix range issue in TestPodSecurityContextAdmission

### PR DESCRIPTION
**Pull request by gaurav1086**
Signed-off-by: Gaurav Singh <gaurav1086@gmail.com>

**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind api-change
 /kind bug
> /kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:
Go uses the same address variable while iterating in a range, so use a copy when using its address.
https://play.golang.org/p/wh5_P3NpZXl

**Which issue(s) this PR fixes**:

Fixes #
N/A
**Special notes for your reviewer**:
N/A
**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
NONE
```

### CODE REVIEW

File: plugin/pkg/admission/securitycontext/scdeny/admission_test.go [hunk 1]
```diff
@@ -139,8 +139,8 @@ func TestPodSecurityContextAdmission(t *testing.T) {
 			errorExpected: true,
 		},
 	}
-	for _, test := range tests {
-		pod.Spec.SecurityContext = &test.securityContext
+	for i, test := range tests {
+		pod.Spec.SecurityContext = &tests[i].securityContext
```
Comment (derekwaynecarr) [inline plugin/pkg/admission/securitycontext/scdeny/admission_test.go:143]: i am missing why this change is needed?

the assignment and validation happens in each iteration of the loop.

Comment (gaurav1086) [reply plugin/pkg/admission/securitycontext/scdeny/admission_test.go:143]: @derekwaynecarr, my bad. Here's the sample illustration. https://play.golang.org/p/wh5_P3NpZXl . Update the PR description as well.

**Comment by gaurav1086** [comment]
cc @liggitt

**Comment by gaurav1086** [comment]
cc @lavalamp

**Comment by lavalamp** [comment]
The code would be wrong if the reference escaped the loop, but it doesn't, so there's no need to change this.

**Comment by gaurav1086** [comment]
@lavalamp thanks for clarification. closing this.
